### PR TITLE
Oracleless ctoken mint and redeem

### PR DESCRIPTION
**Pull request by nope-finance**

### CODE REVIEW

File: token-lending/program/tests/deposit_reserve_liquidity.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@ async fn test_success() {
     );
 
     // limit to track compute unit increase
-    test.set_bpf_compute_max_units(30_000);
+    test.set_bpf_compute_max_units(35_000);
```
Comment (0xkiplet) [inline token-lending/program/tests/deposit_reserve_liquidity.rs]: Is there a way we can assert that the reserve interest has been refreshed in these tests?

**Comment by oxrooter** [comment]
lgtm

**Review by 0xodia — APPROVED** [review]
lgtm

**Review by oxrooter — APPROVED** [review]
lgtm

File: token-lending/program/tests/deposit_obligation_collateral.rs [hunk 2]
```diff
@@ -112,4 +127,6 @@ async fn test_success() {
         user_collateral_balance,
         initial_user_collateral_balance - SOL_DEPOSIT_AMOUNT_LAMPORTS
     );
+
+    assert!(sol_reserve.liquidity.cumulative_borrow_rate_wads > old_borrow_rate);
```
Comment (0xkiplet) [inline token-lending/program/tests/deposit_obligation_collateral.rs:131]: 👍

File: token-lending/program/src/instruction.rs [hunk 3]
```diff
@@ -161,15 +161,15 @@ pub enum LendingInstruction {
...
+    /// Deposit collateral to an obligation.
     ///
     /// Accounts expected by this instruction:
     ///
     ///   0. `[writable]` Source collateral token account.
     ///                     Minted by deposit reserve collateral mint.
     ///                     $authority can transfer $collateral_amount.
     ///   1. `[writable]` Destination deposit reserve collateral supply SPL Token account.
-    ///   2. `[]` Deposit reserve account - refreshed.
+    ///   2. `[writable]` Deposit reserve account.
```
Comment (0xkiplet) [inline token-lending/program/src/instruction.rs:172]: Nit: Do we need to make the same comment change for  `RedeemReserveCollateral` given it no longer needs to be refreshed?

**Review by 0xkiplet — APPROVED** [review]

File: token-lending/program/src/instruction.rs (hunk 3 above)
Comment (nope-finance) [reply token-lending/program/src/instruction.rs:172]: ah yeh prob should change comment (the reserve over there was alright writable so just removing the "refreshed"

**Comment by nope-finance** [comment]
cool gonna merge this and try to test it on beta later today

**Pull request merged**